### PR TITLE
Solves issue: Slack bot does not reconnect when connection get closed

### PR DESCRIPTION
**Pull request by nmorenor**
When the Ping task get an exception and the WebsocketSession is disconnected it does not reconnect, so any other new message/subscription is not listened.

### CODE REVIEW

**Comment by AmosDoan** [comment]
I hope this PR would be merged soon. In my case, there is so frequently exception in the ping task.

**Comment by rampatra** [comment]
I am on holidays for the last couple of weeks. Will review and merge next week. Hope that’s okay.

File: jbot/src/main/java/me/ramswaroop/jbot/core/slack/Bot.java [hunk 1]
```diff
@@ -324,62 +336,67 @@ public void destroy() {
...
+					webSocketSession.sendMessage(new TextMessage(message.toJSONString()));
+				}
+				isException = false;
+			} catch (Exception e) {
+				logger.error("Error pinging Slack. Slack bot may go offline when not active. Exception: ", e);
+				isException = true;
+				if (!isOpen()) {
+					try {
+						manager.stop();
+					} catch (Throwable t) {
```
Comment (rampatra) [inline jbot/src/main/java/me/ramswaroop/jbot/core/slack/Bot.java]: Change to
```suggestion
					} catch (Exception e) {
```
See [this](https://stackoverflow.com/a/581889/1385441) for more info.

File: jbot/src/main/java/me/ramswaroop/jbot/core/slack/Bot.java [hunk 2]
```diff
@@ -324,62 +336,67 @@ public void destroy() {
...
-                    webSocketSession.sendMessage(new TextMessage(message.toJSONString()));
-                }
-                isException = false;
-            } catch (Exception e) {
-                logger.error("Error pinging Slack. Slack bot may go offline when not active. Exception: ", e);
-                isException = true;
-            }
-        }
-
-        boolean isRunning() {
```
Comment (rampatra) [inline jbot/src/main/java/me/ramswaroop/jbot/core/slack/Bot.java]: isRunning() and isException() aren't used anywhere.

File: jbot/src/main/java/me/ramswaroop/jbot/core/slack/Bot.java [hunk 3]
```diff
@@ -324,62 +336,67 @@ public void destroy() {
...
+					startRTMAndWebSocketConnection();
+				}
+			}
+		}
+
+		boolean isConnected() {
+			return webSocketSession.isOpen();
+		}
+
+		boolean isRunning() {
```
Comment (rampatra) [inline jbot/src/main/java/me/ramswaroop/jbot/core/slack/Bot.java]: isRunning() and isException() aren't used anywhere.

**Pull request merged**